tutorial/developing_packages/editable_cmake_packages/say/conanfile.py

```diff
@@ -0,0 +1,46 @@
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout
+
+
+class SayConan(ConanFile):
+    name = "say"
+    version = "1.0"
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "something_custom": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "something_custom": False}
+
+    # Sources are located in the same place as this recipe, copy them to the recipe
+    exports_sources = "CMakeLists.txt", "src/*", "include/*"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def validate(self):
+        # ensure the consumer's options affect this dependency
+        if not self.options.something_custom:
+            raise ConanInvalidConfiguration("The consumer should set something_custom=True")
```

Comment:
This ``validate()`` seems unrelated to the example, I'd say it should be simplified and removed, and the example focused on the custom layout.

Reply:
The point of validate() was to ensure the build of "say" had the flag set.  That flag was set by the consumer.
The original example built "say" without any influence from the consumer.
If you attempted to build this example with the same procedure as the other example (ie conan-install "say" directly), then it should fail.
Perhaps that should be a different example?

Reply:
Re simplification, I got rid of the Debug build.  It was there in the original example.
Can I also get rid of the "beta9" stuff?
I didn't test on Windows, do I really need the `cmake --preset conan-default` windows specialisation?
Seems weird that it is different on Windows.

Reply:
Can you give me a place where it should be moved?
Are you able to help with the HTML linking?

Reply:
> The point of validate() was to ensure the build of "say" had the flag set. That flag was set by the consumer.

I might still be missing something. The option is not used at all for the build, is it? If it doesn't have any effect on the layout, the build or anything, it still doesn't seem related to an example that is focused on the custom source folder. This would be a completely different use case, which is setting options from consumers, but this still doesn't seem related to the current example of the ``layout()``.